### PR TITLE
Followup ghostfat test

### PR DESCRIPTION
**Pull request by hathach**
follow up to #94 to separate native build tests from cross-compile (for target hardware). `NO_TINYUF2_BUILD` is used for this purpose (along with SKIP_NANOLIB, some later ports especially non-ARM may skip this lib as well). NO_TINYUF2_BUILD introduced in recent PR to allow make.mk and rule.mk to build other executable image such as `erase_firmware` (will also migrate self_update later on). 

Eventually, test_ghostfat/main.c run on its own and interact with a small subset of board_api, those that are used by ghostfat.c. 

PS: I really like this and want to move this to its own unit tests folder, maybe later. Got to catch up with iMXRT board release  https://www.adafruit.com/product/4950

### CODE REVIEW

**Comment by henrygab** [comment]
OK, will review....

File: ports/test_ghostfat/main.c [hunk 1]
```diff
@@ -369,10 +369,12 @@ int DumpDiskImage(void) {
     return ERR_NONE;
 }
 
-int test_main(void)
+int main(void)
 {
     int r;
 
+    uf2_init();
```
Comment (henrygab) [inline ports/test_ghostfat/main.c:376]: Very nice!  So much cleaner! :)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -27,13 +27,8 @@ jobs:
     - name: Setup Python
       uses: actions/setup-python@v2
 
-    - name: Setup Node.js
-      uses: actions/setup-node@v1
-
     - name: Checkout code
       uses: actions/checkout@v2
-      with:
-        submodules: 'true'
```
Comment (henrygab) [inline .github/workflows/build.yml:36]: Oh, submodules are not required for this?  I will defer to you, just want to double-check.

**Review by henrygab — APPROVED** [review]
Well done!  Just two minor questions.  I can approve now, as they are so minor.

File: .github/workflows/build.yml (hunk 2 above)
Comment (hathach) [reply .github/workflows/build.yml:36]: yeah, sub-modules basically is the tinyusb and other mcu driver, the native doesn't uses any of those. ci passed is the proof for that. Those mcu is actually very large for some such as nxp and esp-idf (both > 100MB)

File: ports/test_ghostfat/main.c (hunk 1 above)
Comment (hathach) [reply ports/test_ghostfat/main.c:376]: thanks, yup, I think we could move it to its own tests folder later on :)

File: ports/test_ghostfat/main.c [hunk 3]
```diff
@@ -79,7 +79,7 @@ void DumpBuffer(uint64_t startingOffset, const void * sectorBuffer, uint16_t byt
             for (int k = 0; k < 16; k++) {
                 if (0 == k % 8) {
                     printf(" ");
-                }
+                }//------------- Board  -------------//
```
[the remark this answers is not in the document]
Comment (hathach) [reply ports/test_ghostfat/main.c]: oops, I must accidentally paste text with middle mouse. Happen all the time

**Comment by hathach** [comment]
thank you for reviewing

**Pull request merged**